cpp/src/fil/infer.cu

```diff
@@ -138,7 +138,7 @@ __device__ __forceinline__ vec<NITEMS, output_type> tree_leaf_output(tree_type t
   return out;
 }
 
-template <int NITEMS, typename output_type, typename tree_type>
+template <int NITEMS, typename output_type, bool branch_can_be_categorical, typename tree_type>
```

Comment:
I wonder if a shorter name could be used. E.g. `HAS_CATEGORICAL` (shorter, but not completely accurate) or `CATEGORICAL_POSSIBLE`, `CATEGORICAL_SUPPORTED` (the latter two are shorter and definitely more accurate), `CATS_SUPPORTED`, `CATS_POSSIBLE` (provided that `cats` is used as an abbreviation for `categories` consistently).

The same applies to the name of this parameter in other cases.